### ISSUE TITLE
Error When Inlining Between Projects

### ISSUE THREAD

**Issue by WalkerCodeRanger** [oleg-st/InlineMethod.Fody#10]
Using `InlineMethod.Fody` 0.7.3 and Fody 6.8.1 on .NET 8. Under a certain somewhat complex scenario, the error below happens while compiling (reproduction code in [InlineMethodRepro.zip](https://github.com/user-attachments/files/15965024/InlineMethodRepro.zip)). I'm not quite sure what the scenario is because I wasn't able to simplify the code quite enough for it to be obvious.

```plain
Fody: An unhandled exception occurred:
Exception:
Failed to execute weaver C:\Users\jwalker\.nuget\packages\inlinemethod.fody\0.7.3\build\..\weaver\InlineMethod.Fody.dll
Type:
System.Exception
StackTrace:
   at InnerWeaver.ExecuteWeavers() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 219
   at InnerWeaver.Execute() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 106
Source:
FodyIsolated
TargetSite:


Void ExecuteWeavers()
Object reference not set to an instance of an object.
Type:
System.NullReferenceException
StackTrace:
   at Mono.Cecil.ImportGenericContext.MethodParameter(String method, Int32 position) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 73
   at Mono.Cecil.DefaultMetadataImporter.ImportTypeSpecification(TypeReference type, ImportGenericContext context) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 650
   at Mono.Cecil.DefaultMetadataImporter.ImportType(TypeReference type, ImportGenericContext context) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 492
   at Mono.Cecil.DefaultMetadataImporter.ImportTypeSpecification(TypeReference type, ImportGenericContext context) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 640
   at Mono.Cecil.DefaultMetadataImporter.ImportType(TypeReference type, ImportGenericContext context) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 492
   at Mono.Cecil.DefaultMetadataImporter.ImportField(FieldReference field, ImportGenericContext context) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 661
   at Mono.Cecil.DefaultMetadataImporter.ImportReference(FieldReference field, IGenericParameterProvider context) in C:\projects\fody\cecil\Mono.Cecil\Import.cs:line 738
   at InlineMethod.Fody.InlineMethodWeaver.FixInstructions()
   at InlineMethod.Fody.InlineMethodWeaver.Process()
   at InlineMethod.Fody.ModuleWeaver.ProcessCallInstruction(Instruction instruction, MethodDefinition method)
   at InlineMethod.Fody.ModuleWeaver.ProcessMethod(MethodDefinition method)
   at InlineMethod.Fody.ModuleWeaver.Execute()
   at InnerWeaver.ExecuteWeavers() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 179
Source:
Mono.Cecil
TargetSite:
Mono.Cecil.TypeReference MethodParameter(System.String, Int32)
	A		1	
```

**Comment by oleg-st** [comment]
Fixed in 0.7.4

**Comment by WalkerCodeRanger** [comment]
v0.7.4 fixes this error. When I use it in my project, I get a different error. I'll file an issue with that error soon.
